flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/StopWithSavepoint.java

```diff
@@ -132,6 +146,7 @@ public void onLeave(Class<? extends State> newState) {
 
     @Override
     public void cancel() {
+        operationFailureCause = new FlinkException("The job was cancelled.");
```

Comment:
document why we dont wait for the future here

Reply:
not waiting can cause bad things to happen when the SP completes during cancellation